### PR TITLE
SR2021 deploy changes

### PR DESCRIPTION
**Pull request by RealOrangeOne**
Make a few changes needed from trying to provision the SR2021 server:

- ~Add `epel` module, as it's a dependency on provisioning LE certs, apparently~
- Install bzip2, as it's required to untar the python docs

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
> * Add `epel` module, as it's a dependency on provisioning LE certs, apparently

Really? I thought I got it to work without adding that? Specifically it was 6cc698d which I added that let me get past that.

>     * Install bzip2, as it's required to untar the python docs

File: modules/www/manifests/python_docs.pp [hunk 1]
```diff
@@ -11,7 +11,12 @@
   $target_dir = "${$target_root}${archive_name}"
 
   # wget is generally nicer than curl for downloading things.
-  package { ['wget']:
+  package {'wget':
+    ensure => present,
+  }
+
+  # bzip2 is needed to unzip source
+  package {'bzip2':
```
Comment (PeterJCLaw) [inline modules/www/manifests/python_docs.pp:19]: This is valid but can be simplified:
```suggestion
  # bzip2 is needed to unzip some things
  package { ['wget', 'bzip2']: 
```

**Comment by RealOrangeOne** [comment]
I too was skeptical, but adding it made it run properly, and running `puppet module list --modulepath=/etc/puppet/modules` even with your fix still mentioned it.

File: modules/www/manifests/python_docs.pp [hunk 2]
```diff
@@ -28,7 +33,7 @@
           wget -O - https://www.python.org/ftp/python/doc/${version}/${archive_name}.tar.bz2 | tar -xj -C ${target_root} ",
     provider => 'shell',
     creates => $target_dir,
-    require => [Package['wget'],File[$target_root]],
+    require => [Package['wget'],File[$target_root],Package['bzip2']],
```
Comment (PeterJCLaw) [inline modules/www/manifests/python_docs.pp:36]: You can collapse the common cases together here I _think_ (do check this):
```suggestion
    require => [Package['wget', 'bzip2'],File[$target_root]],
```

**Review by PeterJCLaw — COMMENTED** [review]
Thanks!

I don't think we should need `epel` though -- that's aimed at Enterprise Linux, which we are not.

**Comment by PeterJCLaw** [comment]
> adding it made it run properly

Could you clarify this? What was improper about how it ran previously?

**Comment by RealOrangeOne** [comment]
I tried removing `epel` from the server, and got this message when listing modules:

```
Warning: Missing dependency 'puppet-epel':
  'puppet-letsencrypt' (v6.0.0) requires 'puppet-epel' (>= 3.0.1 < 4.0.0)
  'puppet-python' (v5.0.1-rc0) requires 'puppet-epel' (>= 3.0.0 < 4.0.0)
```

**Comment by PeterJCLaw** [comment]
> I tried removing `epel` from the server, and got this message when listing modules:
> 
> ```
> Warning: Missing dependency 'puppet-epel':
>   'puppet-letsencrypt' (v6.0.0) requires 'puppet-epel' (>= 3.0.1 < 4.0.0)
>   'puppet-python' (v5.0.1-rc0) requires 'puppet-epel' (>= 3.0.0 < 4.0.0)
> ```

Yup, that's been the case for a while. It doesn't seem to actually matter though -- I think the issue is that puppet doesn't have a way to specify dependencies which only apply on some platforms.

**Comment by RealOrangeOne** [comment]
When I tried that, it did still apply fine, but it's hard to tell if that means it's not needed.

I'm learning towards just keeping it. Best case it's actually needed, worst case it silences that warning.

**Pull request merged**